bool data1
---
bool result1